livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -382,32 +382,42 @@ class ChannelControllerImpl(
 
     suspend fun runChannelQuery(pagination: QueryChannelPaginationRequest): Result<Channel> {
         // first we load the data from room and update the messages and channel livedata
-        val channel = runChannelQueryOffline(pagination)
+        val queryOfflineJob = scope.async { runChannelQueryOffline(pagination) }
+        // start the online query before queryOfflineJob.await
+        val queryOnlineJob = if (domainImpl.isOnline()) { scope.async { runChannelQueryOnline(pagination) } } else { null }
 
-        // if we are online we we run the actual API call
+        val channelEntityPair = queryOfflineJob.await()
+        if (channelEntityPair != null) {
+            updateLiveDataFromChannelEntityPair(channelEntityPair)
+        }
 
-        return if (domainImpl.isOnline()) {
-            runChannelQueryOnline(pagination)
+        // if we are online we we run the actual API call
+        return if (queryOnlineJob != null) {
+            val response = queryOnlineJob.await()
+            if (response.isSuccess) {
+                updateLiveDataFromChannel(response.data())
+            }
+            response
         } else {
             // if we are not offline we mark it as needing recovery
             recoveryNeeded = true
+            val channel = channelEntityPair?.channel
```

Comment:
If `channelEntityPair` is null, we are returning a `Result(null, null)` that doesn't make sense

Reply:
the offline result has 3 possible state. unknown (null), empty or an actual channel. I don't think the behaviour changed during this refactor.

Reply:
confirmed, this didn't change: https://github.com/GetStream/stream-chat-android-livedata/blob/master/livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt#L383